README.md

```diff
@@ -1,707 +1,1799 @@
+})
+
+const customSelector = customCreateSelector(
+  [inputSelector1, inputSelector2],
+  resultFunc // `resultFunc` will be passed as the first argument to `customMemoize`
+)
+
+customSelector(
+  ...selectorArgs // Will be memoized by `customArgsMemoize`
+)
+```
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+##### Using `memoize` and `...memoizeOptions`
+
+`createSelectorCreator` can be used to make a customized version of [`createSelector`].
+
+The `memoize` argument is a memoization function to replace `defaultMemoize`.
+
+The `...memoizeOptions` rest parameters are zero or more configuration options to be passed to `memoizeFunc`. The selectors `resultFunc` is passed as the first argument to `memoize` and the `memoizeOptions` are passed as the second argument onwards:
+
+```ts
+const customSelectorCreator = createSelectorCreator(
+  customMemoize, // Function to be used to memoize `resultFunc`
+  option1, // `option1` will be passed as second argument to `customMemoize`
+  option2, // `option2` will be passed as third argument to `customMemoize`
+  option3 // `option3` will be passed as fourth argument to `customMemoize`
+)
+
+const customSelector = customSelectorCreator(
+  [inputSelector1, inputSelector2],
+  resultFunc // `resultFunc` will be passed as first argument to `customMemoize`
+)
+```
+
+Internally `customSelector` calls the memoize function as follows:
+
+```ts
+customMemoize(resultFunc, option1, option2, option3)
+```
+
+##### Additional Examples
+
+###### Customize `equalityCheck` for `defaultMemoize`
+
+```js
+import { createSelectorCreator, defaultMemoize } from 'reselect'
+import isEqual from 'lodash.isequal'
+
+// create a "selector creator" that uses lodash.isequal instead of ===
+const createDeepEqualSelector = createSelectorCreator(defaultMemoize, isEqual)
+
+// use the new "selector creator" to create a selector
+const selectSum = createDeepEqualSelector(
+  [state => state.values.filter(val => val < 5)],
+  values => values.reduce((acc, val) => acc + val, 0)
+)
+```
+
+###### Use memoize function from `Lodash` for an unbounded cache
+
+```js
+import { createSelectorCreator } from 'reselect'
+import memoize from 'lodash.memoize'
+
+const hashFn = (...args) =>
+  args.reduce((acc, val) => acc + '-' + JSON.stringify(val), '')
+
+const customSelectorCreator = createSelectorCreator(memoize, hashFn)
+
+const selector = customSelectorCreator(
+  [state => state.a, state => state.b],
+  (a, b) => a + b
+)
+```
+
+</details>
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+<a id="createstructuredselector"></a>
+
+### createStructuredSelector({ inputSelectors }, selectorCreator = createSelector)
+
+<details><summary><b>Description</b></summary>
+
+A convenience function for a common pattern that arises when using `Reselect`.
+The selector passed to a `connect` decorator often just takes the values of its [`input selectors`]
```

Comment:
:hand: let's ditch the `connect` reference, and instead say it simplifies returning an object made up of selector results.

Reply:
I could not agree more. It seems like `createStructuredSelector` was created specifically to be used with `connect` and now that `connect` is legacy API, it seems like `createStructuredSelector` is also kind of a legacy API.

Reply:
Mostly, yeah.  You _could_ use it with `useSelector` to do `const {a, b} = useSelector(someStructuredSelector)`, and that's still a sort of plausible use case.  (Or you could just do `const {a, b} = useSelector(state => ({a: state.a, b: state.b}), shallowEqual), and have basically the same result.